doc/holochain_101/src/zome/read_and_write.md

```diff
@@ -1 +1,154 @@
+
+All Entries have a type. Entries can be "system" Entries, or "app" Entries. System Entries are Entries integral to the proper functioning of Holochain. Importantly, there are system Entries written to each local source chain, at the time of its initialization, or "genesis".
+
+#### System Entries
+These include firstly, the DNA itself. The DNA is written as the first entry into each local source chain. Importantly, this is what gives everyone totaly confidence in everyone elses chains: the fact that you all know you started out by running the same code (otherwise, the hash of the initial DNA entry would be different). There will be more discussion on this later.
+
+The second Entry to be written to the chain is information about the agent using this source chain. That includes their "public key" which is also their address on the network.
+
+There are also entries created relating to the "Capabilities" features of Holochain, but those will be covered elsewhere.
+
+#### App Entries
+Next, there are the "App" Entries. The definitions for these entry types, and how to validate them, are in the hands of developers. When end users are attempting actions which should author data, the data will first be checked against the validation rules supplied by the DNA developer. The possibilities for App Entry type definitions are endless. Defining App Entry types is how you setup the schemas, data structures, and data integrity of your DNA.
+
+#### Entry Sharing
+Not every Entry must be shared with other people running the same DNA. There are two (and may be more later) levels of sharing for App Entries defined: `public` and `private`.
+
+With `public`, all Entries of that type will get propogated to the other peers running the same DNA, so that they will hold a copy as well, and make your data more resilient, and so that it is available even when you're offline.
+
+With `private`, the Entry will never leave your device. In order to still be able to verify your local source chain, the HEADER for that Entry will be propogated to other peers, but the Entry itself will not.
+
+## Data Propogation
+After any chain-modifying functions are called, Holochain will attempt to notify other peers in the network of those actions. Note that it is not a requirement that the device be online or connected to any other networks or nodes for Holochain to be usable! Holochain will announce the new data over the network, by attempting to publish it to peers. As mentioned previously, if the entry is one of a `private` App Entry type, then only the Header will be published, not the entry itself. Publishing of the data is secondary to being able to author data in the first place, so this will be further elaborated in the [entry validation](./entry_validation.md) article.
```

Comment:
```suggestion
After any chain-modifying functions are called, Holochain will attempt to notify other peers in the network of those actions. Note that it is not a requirement that the device be online or connected to any other networks or nodes for Holochain to be usable! Holochain will persist the changes locally to be gossiped to connected peers at a later date. As mentioned previously, if the entry is one of a `private` App Entry type, then only the Header will be published, not the entry itself. Publishing of the data is secondary to being able to author data in the first place, so this will be further elaborated in the [entry validation](./entry_validation.md) article.
```